packages/ui/src/components/Datepicker/Datepicker.tsx

```diff
@@ -109,39 +111,46 @@ const DatepickerRender: ForwardRefRenderFunction<DatepickerRef, DatepickerProps>
     labelClearButton = "Clear",
     showTodayButton = true,
     labelTodayButton = "Today",
-    defaultDate = new Date(),
+    defaultValue = new Date(),
     minDate,
     maxDate,
     language = "en",
     weekStart = WeekStart.Sunday,
     className,
     theme: customTheme = {},
-    onSelectedDateChanged,
+    onChange,
+    label,
+    value,
     ...props
   },
   ref,
 ) => {
   const theme = mergeDeep(getTheme().datepicker, customTheme);
 
-  // Default date should respect the range
-  defaultDate = getFirstDateInRange(defaultDate, minDate, maxDate);
+  const effectiveDefaultValue = useMemo(() => {
```

Comment:
u can inline this directly into line `const [selectedDate, setSelectedDate] = useState<Date | null>` since it will only run one time when the component is mounted